### ISSUE TITLE
Any way to force shortcuts being active or not? Like normal/insert mode?

### ISSUE THREAD

**Issue by inoicouldalwaysturn2u** [Haojen/vimkey#5]
Hi. Great cross-apple-OS app!

On Safari Mac, sometimes the shortcuts activate even when inputting text on a handful of sites so far. Example: https://twitter.com/compose/tweet. The shortcuts keep happening when writing a tweet. Quick check in dev tools. It looks like it is all divs. No input field[s].

I know I vimkey rules can be disabled for a page/site. I still want to use the shortcuts tho. Is there a way to toggle vimkey shortcuts? Like if I could toggle them off while writing a tweet. Then toggle them back on.

Or if I could tell vimkey somehow to assume something like Twitter's divs acting as input fields are input fields. I am happy to add the exceptions somewhere myself. Or to even contribute them to you/the community.

In other words, happy to help if I can and do some amount of scripting or editing.

Thanks.

**Comment by inoicouldalwaysturn2u** [comment]
I used translate to see the other issues. Seems like my issue is related to #3 which is about not being able to escape out of an input with kb shortcuts alone. Have to use mouse to lose focus.

**Comment by Haojen** [comment]
@inoicouldalwaysturn2u Very tank you feedback,Is very useful. In the next version will be fix it.

**Comment by yyx-dev** [comment]
刚安装好插件就发现这个问题了，GitHub左上角搜索框就没法儿按Esc退出🤣，还请尽快解决哦

**Comment by yyx-dev** [comment]
貌似vimkey在Safari的起始页没法使用哦

**Comment by Haojen** [comment]
@inoicouldalwaysturn2u some div element can trigger input behavior bug now is fixed, the new version is waiting for Apple Store review. thank you again. next time I will do more, and make people can join the project.

Here is more about this version update info:

- Add Custom Keyboard binding feature

- Fix dark mode override website styles
- Fix on some website can't scroll bug
- Fix some input window can't detect bug

- Refactor Setting Page
- Improve Dark mode UI

**Comment by Haojen** [comment]
> 貌似vimkey在Safari的起始页没法使用哦

Thank you for feedback, on the Safari default tab page, we don't have access to control. If I find a solution I will be try.

**Comment by Haojen** [comment]
> 刚安装好插件就发现这个问题了，GitHub左上角搜索框就没法儿按Esc退出🤣，还请尽快解决哦

Copy that, thank you again.

**Comment by Haojen** [comment]
@inoicouldalwaysturn2u The new version now is being released, waiting for Apple review. the new version add [gi] and esc blur. here is update note.


Current Update:

Add [gi] key feature, you can use this key to focus the text input area.

Add AB mode custom keybinding. before only supporting AA mode.

Add Experimental Options, now you can experience new features in advance.

Improve ProMotion / 120Hz scrolling experience, make scrolling silky better.

Fix reset setting bug.

Fix help UI can't show custom keybinding bug.

Fix key trigger conflict bugs.

Fix toast UI conflict with some pages.

**Comment by Haojen** [comment]
> > 刚安装好插件就发现这个问题了，GitHub左上角搜索框就没法儿按Esc退出🤣，还请尽快解决哦
> 
> Copy that, thank you again.

新版本已经修复，iOS正在等待苹果审核发布。 macOS 版本已经发布。

更新日志如下
本次更新：

新增[gi]键功能，可以对文本输入区域进行聚焦。

新增AB模式自定义键绑定，之前只支持AA模式。

新增实验功能选项，可以体验新功能。

提升ProMotion / 120Hz滚动体验，让滚动更流畅。

修复重置设置错误

修复按键绑定可能会冲突的问题

修复弹窗样式有时候会受到网页影响的问题
